vnext/Desktop/BeastWebSocketResource.cpp

```diff
@@ -30,6 +30,90 @@ using std::unique_ptr;
 
 namespace Microsoft::React {
 
+#pragma region BeastWebSocketResource
+
+BeastWebSocketResource::BeastWebSocketResource() noexcept {}
+
+#pragma region IWebSocketResource members
+
+void BeastWebSocketResource::Connect(
+    std::string &&urlString,
+    const Protocols &protocols,
+    const Options &options) noexcept {
+  Url url(std::move(urlString));
+
+  if (url.scheme == "ws") {
+    if (url.port.empty())
+      url.port = "80";
+
+    m_concreteResource = std::make_shared<Beast::WebSocketResource>(std::move(url));
+  } else if (url.scheme == "wss") {
+    if (url.port.empty())
+      url.port = "443";
+
+    m_concreteResource = std::make_shared<Beast::SecureWebSocketResource>(std::move(url));
+  }
+
+  m_concreteResource->SetOnClose(std::move(m_closeHandler));
```

Comment:
>std::move [](http://example.com/codeflow?start=33&length=9)

If we use std::move here, then do we expect that the fields are becoming invalid after the Connect call? I.e. we never expect to call Connect the second time.

Reply:
Correct!

Also note, this is the legacy (Boost/Beast) implementation which I keep around for experimentation and performance comparison.

It's by default not even built into the RNW DLL, and will eventually be deleted.
Only keeping it up to date with the `IWebSocket` semantics.